src/vos/vos_obj.c

```diff
@@ -41,10 +41,95 @@ D_CASSERT((uint32_t)VOS_VIS_FLAG_VISIBLE == (uint32_t)EVT_VISIBLE);
 D_CASSERT((uint32_t)VOS_VIS_FLAG_PARTIAL == (uint32_t)EVT_PARTIAL);
 D_CASSERT((uint32_t)VOS_VIS_FLAG_LAST == (uint32_t)EVT_LAST);
 
+static int
+empty_tree_check(daos_handle_t ih, vos_iter_entry_t *entry,
```

Comment:
Maybe a comment that if this function is ever entered it implicitly means the tree is not empty ... took me way to long to figure that out. It might prevent me from feeling dumb in the future :-P

Reply:
done